src/phylim/apps.py

```diff
@@ -237,30 +237,67 @@ class phylim_to_model_result:
         "tree" a cogent3 tree object
     Return:
         model_result
+    Notes:
+        this app support only UNREST (gn) model,
+        and GTR model (or its degenerated forms, like HKY85, JC69, etc)
+        this app assume tree object derived from piqtree
     """
 
     excludes = ["length", "mprobs"]
 
+    def __init__(self, stationarity: bool = True) -> None:
+        self.stationarity = stationarity
+
     def main(self, tree: PhyloNode) -> model_result:
         params = tree.get_edge_vector()[0].params
         mprobs = tree.params["mprobs"]
 
-        predicates = [
-            predicate.parse(k) for k in params.keys() if k not in self.excludes
-        ]
-        submodel = substitution_model.TimeReversibleNucleotide(predicates=predicates)
+        # build predicates, excluding length and mprobs
+        predicates = [k for k in params.keys() if k not in self.excludes]
+
+        # decide model type based on number of rates/predicates
+        if len(predicates) < 6:
+            predicates = [predicate.parse(k) for k in predicates]
+            submodel = substitution_model.TimeReversibleNucleotide(
+                predicates=predicates
+            )
+        else:
+            submodel = self._gn_constructor(predicates)
         lf = submodel.make_likelihood_function(tree, aligned=True)
         lf.set_motif_probs(mprobs)
 
+        # set to stationary as default
+        lf.set_param_rule("mprobs", is_constant=self.stationarity)
+
         result = model_result(
             name=lf.name,
-            source="unknown",
+            source="frompiqtree",
```

Comment:
That value used to `source` should be unique as it gets used to qrite out results/

I would hope that the tree object returned from piqtree records the alignment from which it was derived under `tree.params["source"]` and if it doesn't, then that needs to be added. (I will look there in a second and make an issue if it doesn't.)